nusantara/nusa_datasets/indo4b/indo4b.py

```diff
@@ -0,0 +1,186 @@
+    consists of around 3.6B words, with around 250M sentences."""
+
+    DEFAULT_CONFIG_NAME = "indo4b_source"
+
+    
+    BUILDER_CONFIGS = [
+        NusantaraConfig(
+            name="indo4b_source",
+            version=_SOURCE_VERSION,
+            description="Indo4B source schema",
+            schema="source",
+            subset_id="indo4b",
+        ),
+        NusantaraConfig(
+            name="indo4b_nusantara_ssp",
+            version=_NUSANTARA_VERSION,
+            description="Indo4B Nusantara schema",
+            schema="nusantara_ssp",
+            subset_id="indo4b",
+        ),
+    ]
+
+    def _info(self) -> datasets.DatasetInfo:
+        if self.config.schema == "source":
+            features = datasets.Features(
+                {
+                    "id": datasets.Value("string"),
+                    "text": datasets.Value("string"),
+                }
+            )
+        elif self.config.schema == "nusantara_ssp":
+            features = schemas.self_supervised_pretraining.features
+
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=features,
+            homepage=_HOMEPAGE,
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager) -> List[datasets.SplitGenerator]:
+        """Returns SplitGenerators."""
+
+        url = _URLS["indo4b"]
+        path = dl_manager.download_and_extract(url) + "/processed_uncased_blanklines"
+
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN,
+                gen_kwargs={
+                    "filepath": path,
+                    "split": "train",
+                },
+            ),
+        ]
+
+    def _generate_examples(self, filepath, split: str) -> Tuple[int, Dict]:
+        """Yields examples as (key, example) tuples."""
+
+        counter = 0
+        for txt_path in glob.glob(f'{filepath}/*.txt'):
+            with open(txt_path, encoding="utf-8") as f:
+                if self.config.schema == "source":
+                    for row in f:
+                        if row.strip() != "":
+                            yield (
+                                counter,
+                                {
+                                    "id": str(counter),
+                                    "text": row.strip(),
+                                },
+                            )
+                            counter += 1
+                elif self.config.schema == "nusantara_ssp":
+                    for row in f:
+                        if row.strip() != "":
+                            yield (
+                                counter,
+                                {
+                                    "id": str(counter),
+                                    "text": row.strip(),
+                                },
+                            )
+                            counter += 1
```

Comment:
minor: since it's duplicated (exactly the same) should we just remove the conditional?